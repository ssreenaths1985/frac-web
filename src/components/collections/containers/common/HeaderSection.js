import React from "react";
import BrandSection from "../../../common/BrandSection";
import TopNavBar from "../../../common/TopNavBar";
import InfoNavBar from "../../../common/InfoNavBar";

/**
 * Component which loads the header section
 * of all pages
 */

class HeaderSection extends React.Component {
  _isMounted = false;

  constructor(props) {
    super(props);
    this.receiveClients = this.receiveClients.bind(this);
    this.receiveSessions = this.receiveSessions.bind(this);
  }

  componentDidMount() {
    this._isMounted = true;

    document.addEventListener("click", function (event) {
      if (document.getElementById("dropdownForDeptOne")) {
        document
          .getElementById("dropdownForDeptOne")
          .addEventListener("click", function (event) {
            if (
              event.target.lastChild &&
              event.target.lastChild.id === "deptListOne"
            ) {
              event.stopPropagation();
            } else {
              document.getElementById("deptSearchOne").value = "";
              let li, ul, i, txtValue;
              ul = document.getElementById("deptListOne");
              li = ul.getElementsByTagName("button");

              // Loop through all list items, and hide those who don't match the search query
              for (i = 0; i < li.length; i++) {
                txtValue = li[i].textContent || li[i].innerText;
                if (txtValue.toUpperCase().indexOf("") > -1) {
                  li[i].style.display = "";
                } else {
                  li[i].style.display = "none";
                }
              }
            }
          });
      }
    });
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  receiveClients = (list) => {
    this.props.getClients(list);
  };

  receiveSessions = (data) => {
    this.props.getSessions(data);
  }

  render() {
    return (
      <div className="row">
        <BrandSection />
        <TopNavBar
          pathName={this.props.history.location.pathname}
          history={this.props.history}
          keycloak={this.props.keycloak}
        />
        <InfoNavBar
          {...this.props}
          history={this.props.history}
          keycloak={this.props.keycloak}
          receiveClients={this.receiveClients}
          incomingMsg={this.props.sendData}
          outgoingMsg={this.props.gotData}
          receiveSessions={this.receiveSessions}
        />
      </div>
    );
  }
}

export default HeaderSection;
