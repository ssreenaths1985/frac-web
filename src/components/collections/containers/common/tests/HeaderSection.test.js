import React from "react";
import { shallow } from "enzyme";
import HeaderSection from "../HeaderSection";

const headerProps = {
  history: {
    location: {
      pathname: "/collections",
      search: "",
      hash: "",
      state: null,
      key: "zgfeb7",
    },
  },
};

let wrapped = shallow(<HeaderSection {...headerProps}></HeaderSection>);

describe("HeaderSection", () => {
  it("should render the HeaderSection Component correctly", () => {
    expect(wrapped).toMatchSnapshot();
  });
});
