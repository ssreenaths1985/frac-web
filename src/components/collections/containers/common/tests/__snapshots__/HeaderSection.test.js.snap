// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSection should render the HeaderSection Component correctly 1`] = `
<div
  className="row"
>
  <BrandSection />
  <TopNavBar
    history={
      Object {
        "location": Object {
          "hash": "",
          "key": "zgfeb7",
          "pathname": "/collections",
          "search": "",
          "state": null,
        },
      }
    }
    pathName="/collections"
  />
  <InfoNavBar
    history={
      Object {
        "location": Object {
          "hash": "",
          "key": "zgfeb7",
          "pathname": "/collections",
          "search": "",
          "state": null,
        },
      }
    }
  />
</div>
`;
